### PR TITLE
Myen 549 keymanager integration

### PR DESCRIPTION
**Pull request by jrhender**
- [x] Providing KeyManager WalletConnect option which directly opens KM site in new window/tab.
- [x] Using WalletProvider instead of useMetamask and useEwKM. This is more extensible in the event of future type and keeps selection mutually exclusive
- [x] Can pass in option KM url into `IAM` constructor
- [x] Deprecate `useMetaMask` option in favor of `WalletProvider` enum
- [x] Fix & refactor of `getPublicKey` function. See #100 for details 

**Note**: This PR could be merged in order to deploy to dev environment to test but currently KM login is failing as retrieval of publicKey from signature does not validate

### CODE REVIEW

File: src/iam.ts [hunk 1]
```diff
@@ -125,15 +126,24 @@ export class IAM extends IAMBase {
...
-      useMetamaskExtension: false
+      reinitializeMetamask,
+      walletProvider
+    }: { useMetamaskExtension?: boolean; reinitializeMetamask?: boolean, walletProvider?: WalletProvider } = {
     }
   ): Promise<InitializeData> {
     try {
+      // Deprecated warning below can be removed after param is removed
+      if (useMetamaskExtension !== undefined) {
+        console.warn("Use of useMetamaskExtension parameter is deprecated. Use walletProvider enumeration instead.")
```
Comment (jrhender) [inline src/iam.ts:137]: @dwojno @manihagh Do you agree with this approach of signalling that a param is deprecated to users of `iam-client-lib`

Comment (dwojno) [reply src/iam.ts:137]: @jrhender I think we should just switch to new approach using walletProvider enum, as for now only switchboard is using iam-client-lib so it should be fine. I think leaving it both may cause a lot of mess in initializing connection methods on iam-client-lib, what do you think?

Comment (jrhender) [reply src/iam.ts]: @dwojno Thanks for the advice, I removed the `useMetamaskExtension` param

File: src/iam/iam-base.ts [hunk 2]
```diff
@@ -262,31 +273,58 @@ export class IAMBase {
...
-        bridge: this._connectionOptions.bridgeUrl
+        bridge: this._connectionOptions.bridgeUrl,
+        qrcode: showQRCode
       });
+
+      if (walletProvider === WalletProvider.EwKeyManager) {
+        this._walletConnectProvider.wc.on("display_uri", (err, payload) => {
+          // uri is expected to be WalletConnect Standard URI https://eips.ethereum.org/EIPS/eip-1328
+          const wcUri = payload.params[0];
+          console.log(wcUri);
```
Comment (dwojno) [inline src/iam/iam-base.ts:293]: If not needed please remove console.log 😄

Comment (jrhender) [reply src/iam/iam-base.ts]: Thanks, good catch, I've removed it.

**Review by dwojno — APPROVED** [review]

**Pull request merged**